data-prepper-plugins/parquet-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/parquet/PrimitiveDataTypeChecker.java

```diff
@@ -0,0 +1,40 @@
+package org.opensearch.dataprepper.plugins.codec.parquet;
+
+import org.apache.parquet.example.data.simple.SimpleGroup;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+
+public class PrimitiveDataTypeChecker {
+
+    private PrimitiveDataTypeChecker(){
+
+    }
+
+    public static Object checkPrimitiveDataType(Type field, SimpleGroup simpleGroup, int fieldIndex) {
+        String fieldPrimitiveDataType;
+        if (field.isPrimitive()) {
+            PrimitiveType primitiveType = field.asPrimitiveType();
+            fieldPrimitiveDataType = primitiveType.getPrimitiveTypeName().name();
+            switch (fieldPrimitiveDataType) {
+                case "INT32":
+                    return simpleGroup.getInteger(fieldIndex, 0);
+                case "BOOLEAN":
+                    return simpleGroup.getBoolean(fieldIndex, 0);
+                case "FLOAT":
+                    return simpleGroup.getFloat(fieldIndex, 0);
+                case "DOUBLE":
+                    return simpleGroup.getDouble(fieldIndex, 0);
+                case "INT96":
+                    return simpleGroup.getInt96(fieldIndex, 0);
+                case "INT64":
+                case "LONG":
+                    return simpleGroup.getLong(fieldIndex,0);
+                case "FIXED_LEN_BYTE_ARRAY":
+                    return simpleGroup.getValueToString(fieldIndex, 0);
+                default:
```

Comment:
Similar to the Avro codec, please throw an exception when there is unsupported Parquet data. We can follow-on with full support later. But, when something is not supported we should fail-fast.

Reply:
Done. Exception thrown for unsupported types.

Reply:
Done